src/tribler-core/tribler_core/modules/versioncheck_manager.py

```diff
@@ -29,23 +30,29 @@ async def stop(self):
         await self.shutdown_task_manager()
 
     async def check_new_version(self):
+        for version_check_url in VERSION_CHECK_URLS:
+            if await self.check_new_version_api(version_check_url):
```

Comment:
it is better to wrap the thing in `asyncio.wait_for` and catch timeout error here. Otherwise, the check could get stuck.